rosidl_generator_cpp/resource/msg__struct.hpp.em

```diff
@@ -29,9 +29,12 @@ header_guard_variable = '__'.join([x.upper() for x in header_guard_parts]) + '_'
 #endif
 
 @{
+import em
```

Comment:
I don't think an `em` template should import `em` again to evaluate template strings defined in this template.

Instead you should define a function doing the same as what the template is doing but with plain Python. Depending on the size of that code you might want to define that in the Python modules and only pass a function handle to the template.

Reply:
I ended up doing this, and it sped up the generation time quite a bit.  So thanks for the advice.  The downside to what I did in 84e2a51 is that it now uses "print" again to generate code, which we had decided earlier wasn't great.  I'm not quite sure how to reconcile these two competing goals; additional advice welcome.  Also, I ended up moving much of the python code for the creation of the lists into the pure-python place in b49d699, which cleans up the template quite a bit and allows us to get linting, etc support.

Reply:
Instead of printing within the functions you can return a list of string instead. I would also suggest to not embed the indentation within the function. So the functions return something like this:

```
def func():
    lines = []
    lines.append('some code;')
    ...
    return lines
```

Based on that your template can then apply the right indentation necessary in the context where it is being used:

```
    {
@[for line in generate_zero_string(membset, "_alloc, _init")]@
      @line
@[end for]@
    }
```

Reply:
Done in the latest commit 889cb07